plugins/hydra_nevergrad_sweeper/hydra_plugins/hydra_nevergrad_sweeper/_impl.py

```diff
@@ -0,0 +1,186 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+import logging
+from typing import Any, Dict, List, Optional, Tuple, Union, cast
+
+import nevergrad as ng
+from hydra.core.config_loader import ConfigLoader
+from hydra.core.override_parser.overrides_parser import OverridesParser
+from hydra.core.override_parser.types import (
+    ChoiceSweep,
+    IntervalSweep,
+    Override,
+    Transformer,
+)
+from hydra.core.plugins import Plugins
+from hydra.plugins.launcher import Launcher
+from hydra.plugins.sweeper import Sweeper
+from hydra.types import TaskFunction
+from omegaconf import DictConfig, ListConfig, OmegaConf
+
+from .config import OptimConf, ScalarConfigSpec
+
+# pylint: disable=logging-fstring-interpolation,no-self-used
+log = logging.getLogger(__name__)
+
+
+def verify_scalar_bounds(scalar: ng.p.Scalar) -> None:
+    if scalar.integer:
+        a, b = scalar.bounds
+        if a is not None and b is not None and b - a <= 6:
+            raise ValueError(
+                "For integers with 6 or fewer values, use a choice instead"
+            )
```

Comment:
Why is that?
Is this an actual requirement from Nevergrad? it seems arbitrary.

If so, would it make sense to do this automatically in the plugin instead of surfacing this error to the user?

Reply:
per our offline discussion, removed this check